analyzers/pattern_detector.py

```
@@ -3,178 +1,152 @@
+        """
+        Detect patterns in execution history.
+
+        Returns:
+            Dict[str, Any]: Detected patterns
+        """
+        try:
+            conn = sqlite3.connect(self.db_path)
+            cursor = conn.cursor()
+
+            # Get execution history for the last 7 days
+            seven_days_ago = (datetime.utcnow() - timedelta(days=7)).isoformat()
```

Ruff (0.11.9)

`datetime.datetime.utcnow()` used

(DTZ003)